### ISSUE TITLE
Possible thread leak in "calico-node -felix"

### ISSUE THREAD

**Issue by igcherkaev** [projectcalico/calico#5018]
## Expected Behavior

Thread count should stay about same over time.

## Current Behavior

It appears to be steadily growing.

![image](https://user-images.githubusercontent.com/18666442/138737078-11a6956f-8c70-476e-9e6a-29cd14b8fcba.png)

Overall, we don't see any issues with calico, but our monitoring system raised several alerts (our monitoring team just began adding various checks for our kubernetes nodes) about thread count. It differs node by node, and we have yet to see the pattern, as some nodes appear to be ok, but on some it's growing pretty fast. On one node it's already over 9000!

```bash
rnqkbs401 ~ # ps -eLf | grep calico-node | wc -l
9971
```

Nothing in the calico logs suggests the reason for having so many threads.

## Context

We upgraded from 3.19 to 3.20.2 not a while ago, but we haven't been monitoring thread count on the nodes, so we can't say if it's related to the upgrade or not. At this point we'd like to make sure it's an expected behavior from calico-node.

## Your Environment
* Calico version
```
$ calicoctl version
Client Version:    v3.20.2
Git commit:        dcb4b76a
Cluster Version:   v3.20.2
Cluster Type:      typha,kdd,k8s,bgp,lvp
```
* Orchestrator version (e.g. kubernetes, mesos, rkt): 
```
Kubernetes v1.21.5 and v1.20.9
```
* Operating System and version:
```
Flatcar Container Linux by Kinvolk 2765.2.5 (Oklo)   5.10.38-flatcar   containerd://1.4.4
```

**Comment by caseydavenport** [comment]
Looks like https://github.com/projectcalico/felix/pull/3052 is a candidate fix - @mikestephen do we have confirmation that this fixes this issue?

**Comment by caseydavenport** [comment]
This should be fixed in v3.21, which was released yesterday.
